File: spoj-br/C/FATORIA2/FATORIA2.c
// https://br.spoj.com/problems/QUERM/

#include <stdio.h>

int main()
{
    int N, p, x, t = 1;

    scanf("%d", &N);
    while (N > 0)
    {
        for(p = 1;p <= N; p++)
        {
            scanf("%d", &x);
            if(x == p)
            {
                printf("Teste %d\n", t);
                t++;
                printf("%d\n\n", x);
                //break;
            }
        }
        scanf("%d", &N);
    }
    
    return 0;
}

File: spoj-br/Swift/FATORIA2/FATORIA2.swift
// https://br.spoj.com/problems/QUERM/

var N: Int, t: Int = 1

N = Int(readLine()!)!
while (N > 0) {

    let values = readLine()!.split(separator: " ").map { Int($0)! }
    let i = values.indices.filter { ($0 + 1) == values[$0] }.first!
     
    print("Teste \(t)");
    t = t+1;
    print("\(values[i])\n");  

    N = Int(readLine()!)!
}